akkeeper/src/main/scala/akkeeper/config/RichConfig.scala

```diff
@@ -15,102 +15,133 @@
+      val configContainers = akkeeperConfig.getConfigList("containers").asScala
+      configContainers.map(ContainerDefinition.fromConfig)
     } else {
-      Map.empty
+      Seq.empty
     }
   }
 
-  def getListOfStrings(path: String): Seq[String] = {
-    if (config.hasPath(path)) {
-      config.getStringList(path).asScala
+  // Akka configs
+  def akkaActorSystemName: String = akkaConfig.getString("system-name")
+  def akkaPort: Int = akkaConfig.getInt("port")
+  def akkaSeedNodesNum: Int = akkaConfig.getInt("seed-nodes-num")
+  def akkaJoinClusterTimeout: FiniteDuration = akkaConfig.getDuration("join-cluster-timeout")
+  def akkaLeaveClusterTimeout: FiniteDuration = akkaConfig.getDuration("leave-cluster-timeout")
+
+  // Monitoring configs
+  def monitoringLaunchTimeout: FiniteDuration = monitoringConfig.getDuration("launch-timeout")
+
+  // Launcher configs
+  def launcherTimeout: Option[Duration] = {
+    if (launcherConfig.hasPath("timeout")) {
+      Some(launcherConfig.getDuration("timeout", TimeUnit.SECONDS).seconds)
     } else {
-      Seq.empty
+      None
     }
   }
 
-  def getActorSystemName: String = {
-    config.getString("akkeeper.akka.system-name")
-  }
+  // Zookeeper configs
+  def zookeeperClientConfig: ZookeeperClientConfig = ZookeeperClientConfig.fromConfig(zookeeperConfig)
 
-  def getYarnConfig: Config = {
-    config.getConfig("akkeeper.yarn")
-  }
+  // Rest API configs
+  def restPort: Int = restConfig.getInt("port")
+  def restPortMaxAttempts: Int = restConfig.getInt("port-max-attempts")
+  def restRequestTimeout: Timeout = Timeout(
+    restConfig.getDuration("request-timeout", TimeUnit.MILLISECONDS),
+    TimeUnit.MILLISECONDS)
 
-  def getYarnApplicationName: String = {
-    getYarnConfig.getString("application-name")
-  }
+  // Kerberos configs
+  def kerberosTicketCheckInterval: Long =
+    kerberosConfig.getDuration("ticket-check-interval", TimeUnit.MILLISECONDS)
 
-  def getYarnStagingDirectory(conf: Configuration, appId: String): String ={
+  // Yarn configs
+  def yarnApplicationName: String = yarnConfig.getString("application-name")
+  def yarnJvmArgs: mutable.Seq[String] = yarnConfig.getStringList("master.jvm.args").asScala
+  def yarnMaxAttempts: Int = yarnConfig.getInt("max-attempts")
+  def yarnClientThreads: Int = yarnConfig.getInt("client-threads")
+  def yarnStagingDirectory(conf: Configuration, appId: String): String ={
     val basePath =
-      if (getYarnConfig.hasPath("staging-directory")) {
-        getYarnConfig.getString("staging-directory")
+      if (yarnConfig.hasPath("staging-directory")) {
+        yarnConfig.getString("staging-directory")
       } else {
         new Path(FileSystem.get(conf).getHomeDirectory, ".akkeeper").toString
       }
     new Path(basePath, appId).toString
   }
 
-  def getZookeeperClientConfig: ZookeeperClientConfig = {
-    ZookeeperClientConfig.fromConfig(config.getConfig("akkeeper.zookeeper"))
+  // Yarn master configs
```

Comment:
Just a master, not related to `YARN` per se.

Reply:
Done